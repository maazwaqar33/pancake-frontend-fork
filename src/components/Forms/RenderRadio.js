import React, { Fragment } from 'react';
import ErrorMessage from '../../components/Forms/Error';

const _ = require('lodash');

const isObject = (obj, key) => {
  return _.isObject(obj) ? obj[key] : obj;
};

class RenderRadio extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      openPanel: false,
    };
  }

  render() {
    const { label, isRequired, instructions, options, input } = this.props;
    return (
      <Fragment>
        <fieldset
          className={
            options && options.length > 2 ? 'radio-list' : 'field radio-group'
          }
        >
          {label && (
            <legend>
              {label}
              {isRequired && <span className="aria-hidden">(required)</span>}
            </legend>
          )}
          {instructions && (
            <p dangerouslySetInnerHTML={{ __html: instructions }} />
          )}
          <div>
            <div>
              {options &&
                options.map((item, key) => {
                  return (
                    <label
                      key={key}
                      className={this.props.className && this.props.className}
                    >
                      <input
                        {...input}
                        type="radio"
                        value={isObject(item, 'value')}
                      />
                      <span style={{ border: '1px solid black' }}>
                        {isObject(item, 'label')}
                      </span>
                    </label>
                  );
                })}
            </div>
          </div>
          <ErrorMessage fields={this.props} />
        </fieldset>

        {/* <Checkbox props={this.props.options}/> */}
      </Fragment>
    );
  }
}

export default RenderRadio;
