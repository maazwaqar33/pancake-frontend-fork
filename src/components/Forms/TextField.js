import React from 'react';
import ErrorMessage from '../../components/Forms/Error';
import Accordian from '../../components/Forms/Accordian';
import { underscorize } from '../../helpers/strings';
import NumberField from '../Forms/NumberField';
import { Field } from 'redux-form';

/*
  TextField
  type can be specified in FirstTimeApplication.js if you require number field instead of text
*/

export default class TextField extends React.Component {
  constructor(props) {
    super(props);
    this.state = { show: false };
  }

  toggle() {
    this.setState({
      shown: !this.state.shown,
    });
  }

  getValue() {
    if (this.props.input) {
      if (this.props.input.name === this.props.values[this.props.input.name]) {
        return this.props.values['this.props.input.name'];
      }
      return this.props.input.value;
    }
  }

  render() {
    const prepopulatedValue = this.props.prepopulatedValue
      ? this.props.prepopulatedValue[underscorize(this.props.label)]
      : null;
    const showValue = prepopulatedValue || this.getValue();
    return (
      <fieldset className="field">
        <legend>{this.props.label}</legend>
        {this.props.type === 'number' ? (
          <NumberField {...this.props} value={showValue} />
        ) : (
          <input type="text" {...this.props.input} value={showValue} />
        )}
        {this.props.instructions && (
          <p
            className="instructions"
            dangerouslySetInnerHTML={{ __html: this.props.instructions }}
          />
        )}
        {this.props.checkboxFieldName && (
          <Field
            component={Checkbox}
            label={this.props.checkboxLabel}
            name={this.props.checkboxFieldName}
          />
        )}
        {this.props.accordianText && (
          <div>
            <Accordian
              label={this.props.accordianLabel}
              text={this.props.accordianText}
            />
          </div>
        )}
        <ErrorMessage fields={this.props.meta} />
      </fieldset>
    );
  }
}

const Checkbox = props => {
  return (
    <div className="checkbox-group">
      <div>
        <div className="checkboxes">
          <label>
            <input type="checkbox" {...props.input} />
            <span>{props.label}</span>
          </label>
        </div>
      </div>
    </div>
  );
};
