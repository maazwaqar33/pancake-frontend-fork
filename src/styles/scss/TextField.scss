// .input-group {
//   margin: 40px 0;

//   label {
//     font-size: 18px;
//     font-weight: 500;
//     padding: 0;
//   }

//   > div,
//   > p {
//     flex: 1 0 auto;
//     display: flex;
//     justify-content: center;
//     flex-flow: column;
//     position: relative;
//     margin-top: .5em;

//     > input {
//       flex: 1 0 auto;
//       margin: 0;
//     }
//   }
// }

// .multi-textfield {
//   input {
//     margin: 0;
//   }
// }
