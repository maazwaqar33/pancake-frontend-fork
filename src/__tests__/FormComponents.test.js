import React from 'react';
import renderer from 'react-test-renderer';
import RenderRadio from '../components/Forms/RenderRadio';
import TextField from '../components/Forms/TextField';

const Components = [RenderRadio, TextField];

describe('components', () => {
  'use strict';
  Components.map(item => {
    describe(`<${item} />`, () => {
      it('renders correctly', () => {
        const TagName = item;
        const tree = renderer.create(<TagName />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
