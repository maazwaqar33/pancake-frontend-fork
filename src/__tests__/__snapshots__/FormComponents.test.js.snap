// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <function RenderRadio(props) {_classCallCheck(this, RenderRadio);var _this = _possibleConstructorReturn(this, (RenderRadio.__proto__ || Object.getPrototypeOf(RenderRadio)).call(this,
    props));
    _this.state = {
      openPanel: false };return _this;

  } /> renders correctly 1`] = `
<fieldset
  className="field radio-group"
>
  <div>
    <div />
  </div>
  <div />
</fieldset>
`;

exports[`components <function TextField(props) {_classCallCheck(this, TextField);var _this = _possibleConstructorReturn(this, (TextField.__proto__ || Object.getPrototypeOf(TextField)).call(this,
    props));
    _this.state = { show: false };return _this;
  } /> renders correctly 1`] = `
<fieldset
  className="field"
>
  <legend />
  <input
    type="text"
    value={undefined}
  />
  <div />
</fieldset>
`;
